drivers/gles2/rasterizer_canvas_gles2.cpp

```diff
@@ -2326,23 +2332,15 @@ void RasterizerCanvasGLES2::gl_disable_scissor() const {
 	glDisable(GL_SCISSOR_TEST);
 }
 
-void RasterizerCanvasGLES2::initialize() {
-	RasterizerGLES2::gl_check_errors();
-	RasterizerCanvasBaseGLES2::initialize();
-
-	batch_initialize();
-
+void RasterizerCanvasGLES2::initialize_buffer(GLuint vertex_buffer, GLuint index_buffer) {
 	// just reserve some space (may not be needed as we are orphaning, but hey ho)
-	glGenBuffers(1, &bdata.gl_vertex_buffer);
-
 	if (bdata.vertex_buffer_size_bytes) {
-		glBindBuffer(GL_ARRAY_BUFFER, bdata.gl_vertex_buffer);
+		glBindBuffer(GL_ARRAY_BUFFER, vertex_buffer);
 		glBufferData(GL_ARRAY_BUFFER, bdata.vertex_buffer_size_bytes, nullptr, GL_DYNAMIC_DRAW);
 		glBindBuffer(GL_ARRAY_BUFFER, 0);
 
 		// pre fill index buffer, the indices never need to change so can be static
-		glGenBuffers(1, &bdata.gl_index_buffer);
-		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, bdata.gl_index_buffer);
+		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, index_buffer);
```

Comment:
There's no need to do this when using multiple buffers I think. The index buffer can be reused, only the vertex buffer needs multiple versions.